### ISSUE TITLE
Does Captum support Automatic Mixed Precision?

### ISSUE THREAD

**Issue by linchundan88** [pytorch/captum#1199]
Does Captum support Automatic Mixed Precision? Thanks.

**Comment by craymichael** [comment]
Apologies for the delay here.

Captum does not have a separate `amp=True` / mixed-precision API, but it should work with PyTorch autocast as long as the model forward itself works under autocast. Captum calls the forward function you provide and uses PyTorch autograd, so autocast can be controlled by the caller.

The pattern I would recommend is to wrap the model forward function rather than the whole attribution call:

```python
from captum.attr import IntegratedGradients

def amp_forward(inputs):
    with torch.autocast("cuda", dtype=torch.float16):
        return model(inputs)

ig = IntegratedGradients(amp_forward)
attributions = ig.attribute(inputs, baselines=baselines, target=target)
```

Use the autocast device / dtype appropriate for your setup, for example `torch.autocast("cuda", dtype=torch.float16)`, `torch.autocast("cuda", dtype=torch.bfloat16)`, or `torch.autocast("cpu", dtype=torch.bfloat16)`.

Two caveats: Captum does not guarantee that attribution tensors preserve the autocast dtype, since some methods intentionally accumulate or fit internal quantities in `float32`; and `GradScaler` is usually not relevant for attribution calls because Captum is not doing a training `loss.backward()` step.

I ran a quick local check with CPU `bfloat16` autocast on a small model, and `IntegratedGradients`, `Saliency`, `FeatureAblation`, and `GradientShap` all ran successfully with finite attributions while the model forward ran under autocast. I'll close this for now since this should answer the question, but feel free to follow up if you are seeing a specific method or model fail under autocast.
